cardano-api/src/Cardano/Api/IO.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Cardano.Api.IO
+  ( OutputFile(..)
+
+  , writeByteStringFile
+  , writeByteStringOutput
+
+  , writeLazyByteStringFile
+  , writeLazyByteStringOutput
+
+  , writeTextFile
+  , writeTextOutput
+  ) where
+
+import           Cardano.Api.Error (FileError (..))
+
+import           Control.Monad.Except (runExceptT)
+import           Control.Monad.IO.Class (MonadIO (..))
+import           Control.Monad.Trans.Except.Extra (handleIOExceptT)
+import           Data.Aeson.Types (FromJSON, ToJSON)
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString.Char8 as BS
+import qualified Data.ByteString.Char8 as BSC
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.ByteString.Lazy as LBSC
+import           Data.String (IsString)
+import           Data.Text (Text)
+import qualified Data.Text.IO as Text
+import           GHC.Generics (Generic)
+
+newtype OutputFile = OutputFile
+  { unOutputFile :: FilePath
+  }
+  deriving Generic
+  deriving newtype (Eq, Ord, Show, IsString, ToJSON, FromJSON)
+
+writeByteStringFile :: MonadIO m => FilePath -> ByteString -> m (Either (FileError ()) ())
+writeByteStringFile fp bs = runExceptT $
+  handleIOExceptT (FileIOError fp) $ BS.writeFile fp bs
+
+writeByteStringOutput :: MonadIO m => Maybe FilePath -> ByteString -> m (Either (FileError ()) ())
```

Comment:
The first argument is a bit unclear, it takes a moment to understand that `Nothing` means `stdout`. Maybe encoding that information in the  type would be better? For example:
```haskell
data OutputFileDescriptor
  = OutputFile FilePath
  | OutputStream Handle
```
where `Handle` can be `stdout` or `stderr`   (which doesn't really matter here).

Reply:
I've created: https://github.com/input-output-hk/cardano-node/issues/5062